src/vm/contract.rs

```diff
@@ -327,7 +371,31 @@ impl Ord for WitnessPos {
     fn cmp(&self, other: &Self) -> Ordering {
         assert!(self.timestamp > 0);
         assert!(other.timestamp > 0);
-        self.timestamp.cmp(&other.timestamp)
+        const BLOCK_TIME: i64 = 10 /*min*/ * 60 /*secs*/;
+        match (self.layer1, other.layer1) {
+            (a, b) if a == b => self.height.cmp(&other.height),
```

Comment:
here https://github.com/RGB-WG/rgb-core/issues/117 you said:

> using block height and position in the block of the witness transaction

but this code isn't doing this, is the code incomplete or the documentation old? To me it seems a rule we should keep, to allow ordering of TXs in the same block

Reply:
I do not get: the code is exactly compare the block height, i.e. is doing what is said

Reply:
To me it seems it's comparing just the block height, not the position of the TX in the block

Reply:
Could you please explain how the `OpId` helps ordering TXs mined in the same block? From what I see the `OpId` is the commitment hash of the RGB operation, it seems unrelated to the mining status of the TX.

Reply:
We need to order operations, not transactions. We just utilize anchors and witness transaction to order operations, but we are not locked to just transaction information.

Reply:
Opid is just one of the factors of ordering. You asked "why we do not account for block position of witness transaction" and the answer here is "we do not need it".

Consensus ordering of operations takes many factors (listed in their priority):
1. Blockchain
2. Block height (or timestamp, when compared cross-chain in large window)
3. Operation type (extensions go in front of transitions)
4. Nonce
5. Opid

1-2 are related to witness transactions, and 3-5 to operations themselves. Normally you use nonce (which we discussed a lot with you!) to do definite ordering within the same block! But if you misused nonce and gave two different state transition the same one, than and only opid comes into play. Yes, this is not a meaningful, but at least deterministic. For meaningful ordering one must use nonce properly.